compose/foundation/foundation/src/uikitMain/kotlin/androidx/compose/foundation/text/TextFieldPointerModifier.uikit.kt

```diff
@@ -0,0 +1,210 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.foundation.text
+
+import androidx.compose.foundation.gestures.detectDragGesturesAfterLongPress
+import androidx.compose.foundation.gestures.detectTapGestures
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.text.selection.SelectionAdjustment
+import androidx.compose.foundation.text.selection.TextFieldSelectionManager
+import androidx.compose.foundation.text.selection.getTextFieldSelection
+import androidx.compose.foundation.text.selection.isSelectionHandleInVisibleBound
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.FocusRequester
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.hapticfeedback.HapticFeedbackType
+import androidx.compose.ui.input.pointer.pointerHoverIcon
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.text.TextRange
+import androidx.compose.ui.text.input.OffsetMapping
+import androidx.compose.ui.text.input.TextFieldValue
+
+internal actual fun Modifier.textFieldPointer(
+    manager: TextFieldSelectionManager,
+    enabled: Boolean,
+    interactionSource: MutableInteractionSource?,
+    state: TextFieldState,
+    focusRequester: FocusRequester,
+    readOnly: Boolean,
+    offsetMapping: OffsetMapping
+): Modifier = if (enabled) {
+    if (isInTouchMode) {
+        val selectionModifier = getSelectionModifier(manager)
+        val tapHandlerModifier = getTapHandlerModifier(
+            interactionSource,
+            state,
+            focusRequester,
+            readOnly,
+            offsetMapping,
+            manager
+        )
+        tapHandlerModifier
+            .then(selectionModifier)
+            .pointerHoverIcon(textPointerIcon)
+    } else {
+        Modifier
+            .mouseDragGestureDetector(
+                observer = manager.mouseSelectionObserver,
+                enabled = enabled
+            )
+            .pointerHoverIcon(textPointerIcon)
+    }
+} else {
+    Modifier
+}
+
+@OptIn(InternalFoundationTextApi::class)
+private fun getTapHandlerModifier(
+    interactionSource: MutableInteractionSource?,
+    state: TextFieldState,
+    focusRequester: FocusRequester,
+    readOnly: Boolean,
+    offsetMapping: OffsetMapping,
+    manager: TextFieldSelectionManager
+) = Modifier.pointerInput(interactionSource) {
+    /*
+    We need to move tap recognizer here from selection modifier (as it is in common)
+    because otherwise we have onDoubleTap call and onTap call, and onDoubleTap will execute before onTap.
+     */
+    detectTapGestures(
+        onDoubleTap = { touchPointOffset ->
+            tapTextFieldToFocus(
```

Comment:
We should remove `tapTextFieldToFocus` after applying the next suggestion

Reply:
I guess no because this is for requesting focus on the text field, and the previous suggestions were about tap detecting logic.
Could you explain why?

Reply:
Or maybe should I rename that method to something like "requestTextFieldFocus"?

Reply:
> Could you explain why?

When we handle press, it always triggered before double tap

Reply:
>  further it works like this:

Also rechecked in on desktop, and indeed, it calls 2 times on double tap 🤔. It makes `onPress` not useful for our case.

The delay for `onTap` is not good for us also. The delay is added when we start to handle double tap:
<img width="523" alt="image" src="https://github.com/JetBrains/compose-multiplatform-core/assets/5963351/a3a01cf3-a125-416e-89a5-9a0f661c2a8d">

So, using a custom handler is our only solution here.